core/io/resource_format_binary.cpp

```diff
@@ -579,7 +600,9 @@ Error ResourceLoaderBinary::parse_variant(Variant &r_v) {
 			array.resize(len);
 			Color *w = array.ptrw();
 			if (sizeof(Color) == 16) {
-				f->get_buffer((uint8_t *)w, len * sizeof(real_t) * 4);
+				// Colors always use `float` even with double-precision support enabled
+				static_assert(sizeof(Color) == 4 * sizeof(float));
```

Comment:
There's `if (sizeof(Color) == 16)` before so this would never be true.

Reply:
This is a `static_assert`, it comes up at compile-time. Maybe we can remove the `if` then?